### PR TITLE
Apim 5973 paginated log list for application

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-5973

## Description

Show list of logs (default last 24 hours) + Allow user to paginate list

https://github.com/user-attachments/assets/ffa7dafa-bf8e-42de-9a4d-11fb0534d425

No logs found:

Screenshot 2024-08-05 at 18 21 30

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-cqzrqdhhov.chromatic.com)

### CODE REVIEW

File: gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.html [hunk 1]
```diff
@@ -1,18 +1,98 @@
...
+    <div class="m3-body-medium" i18n="@@applicationLogsExplanation">
+      Logs from all APIs for which this application has an accepted, paused, closed, or resumed subscription. Your administrator may
+      restrict what data you see below.
+    </div>
+    <table mat-table [dataSource]="logs" class="application-logs__table">
+      <ng-container matColumnDef="api">
+        <th mat-header-cell *matHeaderCellDef i18n="@@applicationLogsTableColumnApi" class="m3-title-medium">API</th>
+        <td mat-cell *matCellDef="let log">
+          <div>{{ log.apiName }}</div>
+          <div><span i18n="@@applicationLogsTableVersionV">v</span>{{ log.apiVersion }}</div>
```
Comment (phiz71) [inline gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.html]: Why did you add this `v`? An api publisher could have decided to use a `V_` in the version.
It would end with : `vV_2.6.3`

File: gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.ts [hunk 2]
```diff
@@ -13,13 +13,127 @@
...
+      hasPreviousPage: this.currentLogsPage() > 1,
+      hasNextPage: this.currentLogsPage() < totalPages,
+      currentPage: this.currentLogsPage(),
+      totalPages,
+    };
+  });
+
+  displayedColumns: string[] = ['api', 'timestamp', 'httpMethod', 'responseStatus'];
+
+  private currentLogsPage: WritableSignal<number> = signal(1);
```
Comment (phiz71) [inline gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.ts:94]: Should we read the `page` value from a query param in the activated route?

**Review by phiz71 — CHANGES_REQUESTED** [review]

File: gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.html (hunk 1 above)
Comment (jourdiw) [reply gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.html]: Figma 🤷‍♀️ . I can change it to `Version: ` instead.

File: gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.ts (hunk 2 above)
Comment (jourdiw) [reply gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-tab-logs.component.ts:94]: Yes. I'll do that refactor.

**Review by phiz71 — APPROVED** [review]
